File: R/ames_houses/dslc_documentation/functions/cleanAmesData.R
# Cleaning function for the Ames housing data

# Clean the Ames data
cleanAmesData <- function(ames_data) {
  
  # Replace 'blank' categorical entries with `NA`
  ames_data <- ames_data |>
    mutate_if(is.character, 
              function(.x) if_else(.x == "",  
                                   true = as.character(NA), 
                                   false = .x))
  
  # Convert the character variables to factors.
  ames_data <- ames_data |>
    mutate_if(is.character, as.factor)
  
  # impute missing garage year with house built year
  ames_data <- ames_data |> 
    mutate(Garage.Yr.Blt = if_else(!is.na(Garage.Yr.Blt), 
                                     Garage.Yr.Blt, 
                                     Year.Built))
  
  # replace `Year.Remod.Add` value of 1950 with the `Year.Built` value
  ames_data <- ames_data |> 
    mutate(Year.Remod.Add = if_else(Year.Remod.Add == 1950, 
                                    Year.Built, 
                                    Year.Remod.Add))
  
  # create a date variable
  ames_data <- ames_data |>
    unite(date, Mo.Sold, Yr.Sold, 
          sep = "/", remove = FALSE) |>
    mutate(date = parse_date(date, "%m/%Y"))
  
  # clean the column names
  ames_data <- clean_names(ames_data)
  
  return(ames_data)
}

File: python/ames_houses/dslc_documentation/functions/clean_ames_data.py
# Cleaning function for the Ames housing data
import numpy as np
import pandas as pd

# Clean the Ames data
def clean_ames_data(ames_data):
  
  ames_data_clean = ames_data.copy() 
  
  # impute missing garage year with house built year
  ames_data_clean["Garage Yr Blt"] = np.where(ames_data_clean["Garage Yr Blt"].isna(), 
                                              ames_data_clean["Garage Yr Blt"],
                                              ames_data_clean["Year Built"])
  
  # replace `Year Remod Add` value of 1950 with the `Year Built` value
  ames_data_clean["Garage Yr Blt"] = np.where(ames_data_clean["Year Remod/Add"] == 1950, 
                                              ames_data_clean["Year Built"],
                                              ames_data_clean["Year Remod/Add"])
  
  # create a date variable
  ames_data_clean["date"] = pd.to_datetime(dict(year=ames_data_clean["Yr Sold"], 
                                                month=ames_data_clean["Mo Sold"], 
                                                day=1))
  
  # clean the column names
  ames_data_clean.columns = ames_data_clean.columns.str.lower() \
                                                    .str.replace(' ', '_') \
                                                    .str.replace('/', '_')
  
  ames_data_clean = ames_data_clean.set_index("pid")
  
  return(ames_data_clean)


